internal/config/dir.go

```diff
@@ -0,0 +1,83 @@
+package config
+
+import (
+	"errors"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+)
+
+// HomeDir returns the user's home directory, which can be overridden with the {ENV_PREFIX}HOME variable.
+func (c *Config) HomeDir() (string, error) {
+	d := os.Getenv(c.Application.EnvPrefix + "HOME")
+	if d != "" {
+		return d, nil
+	}
+	return os.UserHomeDir()
+}
+
+// WritableUserDir returns the path to a writable user-level directory.
+func (c *Config) WritableUserDir() (string, error) {
+	if c.writableUserDir != "" {
+		return c.writableUserDir, nil
+	}
+	hd, err := c.HomeDir()
+	if err != nil {
+		return "", err
+	}
+	path := filepath.Join(hd, c.Application.WritableUserDir)
+	if err := mkDirIfNotExists(path); err != nil {
+		return "", err
+	}
+	c.writableUserDir = path
+
+	return path, nil
+}
+
+// TempDir returns the path to a user-specific temporary directory.
+//
+// It creates the temporary directory if it does not already exist.
+//
+// It does not use os.TempDir on Linux as that usually returns /tmp which could
```

Comment:
Perhaps it should use /tmp when the other options are not available (writable), e.g. on a platform.sh container environment.